### ISSUE TITLE
Support worker multiplexing and multiplex sandboxing

### ISSUE THREAD

**Issue by jadenPete** [bazelbuild/rules_scala#1662]
**This issue is part of a broader proposal to merge some of the features in [lucidsoftware/rules_scala](https://github.com/lucidsoftware/rules_scala) into this ruleset.**

[The tracking issue for this effort](https://github.com/bazelbuild/rules_scala/issues/1658)

[The relevant proposal section](https://docs.google.com/document/d/12BmUPpJpworA9ep2P_J00TxXwFRBZDQttUCsoYo0Yao/edit?tab=t.0#heading=h.5g9rdoh86mi).

From the [documentation](https://bazel.build/remote/multiplex) on multiplex workers:

> *Multiplex workers* allow Bazel to handle multiple requests with a single worker process. For multi-threaded workers, Bazel can use fewer resources to achieve the same, or better performance. For example, instead of having one worker process per worker, Bazel can have four multiplexed workers talking to the same worker process, which can then handle requests in parallel. For languages like Java and Scala, this saves JVM warm-up time and JIT compilation time, and in general it allows using one shared cache between all workers of the same type.

Although multiplex workers are an experimental feature, we've been using them in [lucidsoftware/rules_scala](https://github.com/lucidsoftware/rules_scala) for years to great effect and believe the mainline ruleset would benefit significantly from them.

**Comment by crt-31** [comment]
I don't believe rules_scala should depend on _experimental_ features.  But I'd be fine if rules_scala supported multiplex and non-multiplex workers. As long as I can still run builds in a natural way with --workermultiplex=false.

**Comment by jadenPete** [comment]
That makes sense.
